Sources/Basic/Path.swift

```diff
@@ -485,6 +499,16 @@ extension AbsolutePath {
     public func contains(_ other: AbsolutePath) -> Bool {
         return self.components.starts(with: other.components)
     }
+
+    /// Checks the path and throws a `PathError` if path is invalid.
+    ///
+    /// This method is strictly syntactic and does not access the file system
+    /// in any way.
+    public static func validate(path: String) throws {
+        if path.first == "~" {
+            throw PathError.startsWithTilda(path: path)
+        }
```

Comment:
I think we should first check if the string beings with `/` and then perform diagnostic checks like `~`. Something like:

```swift
public static func validate(pathString path: String) throws {
    if path.first == "/" {
        return
    }
   
    if path.first == "~" {
        throw PathError.startsWithTilda(path)
    }

    throw PathError.invalidAbsolutePath(path)
}
```

Reply:
yes, and we can pattern-match